### DISCUSSION TITLE
libzkgroup.dll not found

### DISCUSSION THREAD

**Discussion by Tsjippy** [AsamK/signal-cli#496] [Q&A]
Thanks for this great libray.

I want to use the libzkgroup.dll on my windows system, but it is not found. I put the file in three places:

1. C:\Program Files (x86)\Common Files\Oracle\Java\javapath (defined in my %PATH% variable
2. C:\Program Files\Common Files\Oracle\Java\javapath  %PATH% variable
3.  ...\signal-cli\lib  (signal-protocol-java-2.8.1.jar folder)

However it still sais: 

> WARN App - WARNING: Support for new group V2 is disabled, because the required native library dependency is missing: libzkgroup

What am I doing wrong?

Sending messages is going ok, but not to V2 groups

**Comment by Atomic-Bean** [comment]
When you say `%PATH%` do you mean the OS environment paths or `java.library.path`? signal-cli will only search the latter. [This wiki page](https://github.com/AsamK/signal-cli/wiki/Provide-native-lib-for-libsignal) can help find it for you.

**Comment by Tsjippy** [comment]
I mean OS path which is the same as java.library.path on windows.
To be sure I even added -Djava.library.path= pointing to the libraryfolder but it doesn’t work

**Comment by Atomic-Bean** [comment]
> I mean OS path which is the same as java.library.path on windows.
> To be sure I even added -Djava.library.path= pointing to the libraryfolder but it doesn’t work

Oh, that's interesting. I've never tried setting it up on Windows. I am not sure why exactly it would be messing up.

Maybe you could try insert the `dll` into the `jar` in the same place the former `so` file was. It didn't work for me on my mac, but maybe it will work for you. To add the dll, I think it will be easiest to rename the jar to zip and use a tool to add the file to the zip, then rename back to jar.

**Comment by Tsjippy** [comment]
I extracted signal-cli-0.7.4.jar, signal-metadata-java-0.1.2.jar, signal-protocol-java-2.8.1.jar, signal-service-java-2.15.3_unofficial_16.jar and zkgroup-java-0.7.0.jar. I added libzkgroup.dll to all of them then zipped them to a their jar files.
Still the same error. I noticed the zkgroup jar is version 0.7 while the newest is 0.7.1

**Comment by Tsjippy** [comment]
I just tested it on a rpi, I build a libzkgroup.so file, added it to the root of the zkgroup-java-0.7.0.jar file. There it works. But not on windows

**Comment by Atomic-Bean** [comment]
> I just tested it on a rpi, I build a libzkgroup.so file, added it to the root of the zkgroup-java-0.7.0.jar file. There it works. But not on windows

`so` is for linux based systems so it makes sense that worked on the rpi whereas windows uses dlls. You don't need to add the lib file to all of those jars though, just the zkgroup one, but even so it sounds like that won't work in your case. I'm not sure what the issue is, maybe Asamk has some ideas.

**Comment by Tsjippy** [comment]
Maybe the doll itself is not correct? But I don’t know how to rebuild it

**Comment by Tsjippy** [comment]
Just to be sure I compiled the zkgroup.dll myself instead of using the one from the releases page. But that does not help
